regression/cbmc-java/PhiMergeConstantPropogation/test.desc

```diff
@@ -0,0 +1,7 @@
+CORE 
+Main.class
+--function Main.phiMerge --show-vcc
+activate-multi-line-match
+^.*dynamic_object2#2 == \{ \.@class_identifier="java::Main\$Temp",\s*\.@lock=false \}$
```

Comment:
Should probably check for the other dynamic object too (the instance of `Temp` and the instance of `Ephemeral`)

Reply:
Decided to change the logic to make it more robust: it now just makes sure there's no gen zero objects in guards, doesn't matter where they are.